studio/src/editors/merch-card-collection-editor.js

```diff
@@ -0,0 +1,862 @@
+
+        // Determine which side has more space (left or right of the editor)
+        const spaceOnRight = viewportWidth - editorRect.right;
+        const spaceOnLeft = editorRect.left;
+
+        const position = {
+            // Align with the top of the trigger for vertical positioning
+            top: triggerRect.top,
+            // Center the preview in the available space outside the editor
+            ...(spaceOnRight > spaceOnLeft
+                ? {
+                      left: editorRect.right + spaceOnRight / 2 - 150, // Assuming preview width ~300px
+                      right: undefined,
+                  }
+                : {
+                      right:
+                          viewportWidth -
+                          editorRect.left +
+                          spaceOnLeft / 2 -
+                          150,
+                      left: undefined,
+                  }),
+        };
+
+        this.previewItem = fragment;
+        this.renderPreviewInLightDOM(position, fragment);
+    }
+
+    // Handle preview hide
+    hideItemPreview() {
+        // Remove the preview element from the light DOM
+        if (
+            this.previewElement &&
+            document.body.contains(this.previewElement)
+        ) {
+            document.body.removeChild(this.previewElement);
+            this.previewElement = null;
+        }
+    }
+
+    // Render the preview in the light DOM
+    async renderPreviewInLightDOM(position, previewItem) {
+        // Remove any existing preview element
+        if (
+            this.previewElement &&
+            document.body.contains(this.previewElement)
+        ) {
+            document.body.removeChild(this.previewElement);
+        }
+        // Get the fragment reference for this item
+        const fragmentStore = this.fragmentReferencesMap.get(previewItem.path);
+        if (!fragmentStore) return;
+
+        const fragment = fragmentStore.get();
+        if (!fragment) return;
+
+        const container = document.createElement('div');
+        container.innerHTML = `
+            <div class="preview-container">
+                <div class="preview-backdrop"></div>
+                <div class="preview-popover" style="${position.left !== undefined ? `left: ${position.left}px` : `right: ${position.right}px`}">
+                    <div class="preview-content">
+                        <merch-card>
+                            <aem-fragment ims author fragment="${previewItem.id}"></aem-fragment>
+                        </merch-card>
+                        <sp-progress-circle class="preview" indeterminate size="l"></sp-progress-circle>
+                    </div>
+                </div>
+            </div>
+        `;
+
+        // Add to document body
+        document.body.appendChild(container);
+
+        // Store reference to the container element
+        this.previewElement = container;
+        await container.querySelector('aem-fragment').updateComplete;
```

Comment:
I think this can cause a memory leak if the user moves the mouse quickly over multiple preview triggers. If a preview starts rendering (container A) and the user moves the mouse to another element before the rendering completes, a second preview starts (container B) and `this.previewElement` now points to container B, leaving container A "orphaned" and so when the first async operation completes, it has no way to know it should be cancelled and neither clean-up process can remove container A.
Ideally there should be a mechanism to track and cancel in-progress operations when new ones start.

Reply:
when an element is disconnected from DOM, the events it dispatches are not propagated in DOM.
And as they are no more referenced, they should be GCed automatically.

You can test it actually at, it is very smooth:
https://mwpw-168446--mas--adobecom.aem.live/studio.html#page=content&path=acom&tags=mas%3Astudio%2Fcontent-type%2Fmerch-card-collection

Reply:
here from the profiler standpoint, when different items are rolled over consecutively.  No linear increase.

<img width="1105" alt="image" src="https://github.com/user-attachments/assets/e504df2d-813b-467e-9046-487dbafd9f04" />